### PR TITLE
[ANCHOR-356] SEP-6: Implement withdraw

### PR DESCRIPTION
**Pull request by philipliu**
### Description

This PR implements the GET `/withdraw` endpoint and the reference server implementation so that the E2E tests pass. It also includes minor refactoring to the payment observer.

### Context

SEP-6 implementation.

### Testing

`./gradlew test`

### Known limitations

N/A

### CODE REVIEW

File: api-schema/src/main/java/org/stellar/anchor/api/sep/AssetInfo.java [hunk 1]
```diff
@@ -15,10 +15,17 @@ public class AssetInfo {
   String issuer;
 
   public String getAssetName() {
-    if (issuer != null) {
-      return schema + ":" + code + ":" + issuer;
+    return schema + ":" + makeAssetName(code, issuer);
+  }
+
+  public static String makeAssetName(String assetCode, String assetIssuer) {
```
Comment (lijamie98) [inline api-schema/src/main/java/org/stellar/anchor/api/sep/AssetInfo.java:21]: By reading the names, the purposes of these functions seem unclear to me. Can we merge or comment?

File: api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/GetWithdrawRequest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * The request body of the GET /withdraw endpoint.
+ *
+ * @see <a
+ *     href="https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0006.md#request-1">GET
+ *     /withdraw</a>
+ */
+@Builder
+@Data
+public class GetWithdrawRequest {
```
Comment (lijamie98) [inline api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/GetWithdrawRequest.java:17]: I understand the SEP-6 uses `GET` method. But it is confusing in the service layer. Can we rename it to something less of a "GETTER" name?

File: platform/src/main/java/org/stellar/anchor/platform/component/sep/SepBeans.java [hunk 3]
```diff
@@ -92,6 +92,8 @@ public FilterRegistrationBean<Filter> sep10TokenFilter(JwtService jwtService) {
     registrationBean.setFilter(new Sep10JwtFilter(jwtService));
     registrationBean.addUrlPatterns("/sep6/deposit*");
     registrationBean.addUrlPatterns("/sep6/deposit/*");
+    registrationBean.addUrlPatterns("/sep6/withdraw*");
+    registrationBean.addUrlPatterns("/sep6/withdraw/*");
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/component/sep/SepBeans.java:96]: Do these patterns exist?

File: platform/src/test/kotlin/org/stellar/anchor/platform/service/PaymentOperationToEventListenerTest.kt [hunk 4]
```diff
@@ -510,6 +500,118 @@ class PaymentOperationToEventListenerTest {
...
+  fun `test SEP-6 onReceived with sufficient payment patches the transaction`(
+    assetType: String,
+    assetCode: String,
+    assetIssuer: String?
+  ) {
+    val transferReceivedAt = Instant.now()
+    val transferReceivedAtStr = DateTimeFormatter.ISO_INSTANT.format(transferReceivedAt)
+    val asset = createAsset(assetType, assetCode, assetIssuer)
+
+    val p =
```
Comment (lijamie98) [inline platform/src/test/kotlin/org/stellar/anchor/platform/service/PaymentOperationToEventListenerTest.kt:520]: Rename to something more readable?

**Review by lijamie98 — APPROVED** [review]
Please feel free to merge after taking care of the comments.

File: api-schema/src/main/java/org/stellar/anchor/api/sep/AssetInfo.java (hunk 1 above)
Comment (philipliu) [reply api-schema/src/main/java/org/stellar/anchor/api/sep/AssetInfo.java:21]: I'll update the method names and add Javadoc.

File: api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/GetWithdrawRequest.java (hunk 2 above)
Comment (philipliu) [reply api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/GetWithdrawRequest.java:17]: Sure, I can update them outside of this PR.

File: platform/src/main/java/org/stellar/anchor/platform/component/sep/SepBeans.java (hunk 3 above)
Comment (philipliu) [reply platform/src/main/java/org/stellar/anchor/platform/component/sep/SepBeans.java:96]: Can you explain what you mean?

**Pull request merged**